### ISSUE TITLE
Model performance decreases after fine-tuning using pre-trained weights

### ISSUE THREAD

**Issue by HoranChen** [timbroed/HRFuser#10]
Hi！Thanks for sharing.

But I meet some problem when I fine-tuning using the pre-trained weights.

The model performance decreases to 

**Overall AP@easy, moderate, hard:
bbox AP:about 50, 35,35**

after 1 epoch in STF val dataset

and after few epoch, these would be 

**Overall AP@easy, moderate, hard:
bbox AP:about 10, 5,5**

Why?

I've only changed some parameters in `cascade_rcnn_hrfuser_t_1x_stf_r1248_4mod.py`

```
# AdamW optimizer, no weight decay for position embedding & layer norm
optimizer = dict(
    _delete_=True,
    type='AdamW',
    **lr=0.000000001,(here)**
    betas=(0.9, 0.999),
    weight_decay=0.01,
    paramwise_cfg=dict(
        custom_keys={
            'absolute_pos_embed': dict(decay_mult=0.),
            'relative_position_bias_table': dict(decay_mult=0.),
            'norm': dict(decay_mult=0.)
        }))
runner=dict(max_epochs=20)
lr_config=dict(policy='step', step=[40,50])
data=dict(samples_per_gpu= 1, workers_per_gpu= 2)
seed=0
```

**Comment by timbroed** [comment]
Hi, I don't know what exactly leads to this behavior with your setup. Generally for fine-tuning, you could try these tricks: use a learning rate warmup, use a low lr and freeze, or reduce the lr for the earlier layers of the model (maybe even the whole backbone).
hope this helps

**Comment by HoranChen** [comment]
I have solved the above problem, which should be caused by the batch size being too small.

However, I also have some problems with the following.

1. When I used [the pre-training weights](https://drive.google.com/drive/folders/1e0esMyGVJndVcjbwkyUo29KlyZBtnh1f?usp=sharing), it couldn't achieve the performance described in **SeeingThroughFog** dataset in the paper.
![image](https://github.com/user-attachments/assets/56baf9bb-e340-46a5-a5de-8f9bdaa2f285)
only have
`**test_clear** Overall AP@easy, moderate, hard:
bbox AP:82.8825, 79.5795, 74.8322
**light_fog** Overall AP@easy, moderate, hard:
bbox AP:86.4843, 83.4600, 81.1983
**dense_fog** Overall AP@easy, moderate, hard:
bbox AP:75.4685, 69.7480, 68.4479
**snow** Overall AP@easy, moderate, hard:
bbox AP:84.5524, 78.7630, 75.3420
**rain** Overall AP@easy, moderate, hard:
bbox AP:71.8056, 73.2262, 70.1296`

2. How to use the gated images in SeeingThroughFog dataset? In the .pkl file you provided, I couldn't find the information about the gated images.

Looking forward to your early reply, thanks!

**Comment by timbroed** [comment]
If you are using the the correct checkpoint and the script for all input modalities you should be able to reproduce the results without any training. 

The .pkl files you mentioned are only the converted dense_info files. These files should be placed in the dense folder, which already contains the gated camera images. Please follow steps 1-3 outlined [here](https://github.com/timbroed/HRFuser?tab=readme-ov-file#seeing-through-fog) to ensure the setup is correct.
